tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -65,6 +67,130 @@ public:
    std::unique_ptr<ROOT::Internal::RPageSource> CreatePageSource() const;
 };
 
+class RNTupleProcessorEntry {
+   friend class RNTupleSingleProcessor;
+   friend class RNTupleChainProcessor;
+   friend class RNTupleJoinProcessor;
+   template <typename T>
+   friend class RNTupleProcessorValuePtr;
+
+private:
+   std::unique_ptr<ROOT::REntry> fEntry;
+   bool fIsValid = true;
+
+   RNTupleProcessorEntry(std::unique_ptr<ROOT::REntry> entry) : fEntry(std::move(entry)) {}
+
+   void SetValid() { fIsValid = true; }
```

Comment:
Any reason not to prefer a `SetValid(bool valid)`? I'd say that generally it's a better API as it composes better (e.g. avoids situations where one has to write `if (valid) { SetValid() } else { SetInvalid() }`).